### PR TITLE
Remove Storext gem

### PR DESCRIPTION
**Pull request by citizen428**
## What type of PR is this? (check all applicable)

- [X] Refactor

## Description

This PR is an experiment. While we have two models that include `Storext.model` (`Article` and `User`) neither of them seems to define a `store` or `store_attributes` so it looks like this gem may actually be unused. Somewhat related to the Rails 7 updated (#15825) as this is one of the gems that doesn't have a released version for Rails 7 yet.

## QA Instructions, Screenshots, Recordings

All specs should be green

### UI accessibility concerns?

No UI changes

## Added/updated tests?

- [X] No, and this is why: I wasn't sure if anything needs updating

## [Forem core team only] How will this change be communicated?

- [X] This change does not need to be communicated, and this is why not: minor change, removing unused dependency

### CODE REVIEW

**Comment by citizen428** [comment]
Seems like all gems still pass with the gem removed and the models updated. @benhalpern Do you have any more context around this by any chance?

**Comment by jeremyf** [comment]
tl;dr - The PR lgtm

Reading notes, I found the following references to `store_attribute`, but those are related to the [store_attribute gem](https://github.com/palkan/store_attribute) and not `storext`.

https://github.com/forem/forem/blob/973ff20eb88eb2fa351757d7075f9bea29968556/app/models/profile.rb#L39-L49

https://github.com/forem/forem/blob/973ff20eb88eb2fa351757d7075f9bea29968556/app/services/profile_fields/remove.rb#L34-L41

https://github.com/forem/forem/blob/973ff20eb88eb2fa351757d7075f9bea29968556/app/services/users/update.rb#L107-L116

**Comment by djuber** [comment]
yes, store_attributes is definitely used for profile fields (and that's what those calls are), agreeing and confirming what you see @jeremyf

File: Gemfile.lock [hunk 1]
```diff
@@ -867,11 +855,6 @@ GEM
     view_component (2.47.0)
       activesupport (>= 5.0.0, < 8.0)
       method_source (~> 1.0)
-    virtus (1.0.5)
```
Comment (djuber) [inline Gemfile.lock:870]: ooh, is this the last dependency on virtus? 

I assume all the other gems moved to dry.rb gems, since the author of virtus openly deprecated it in favor of  dry-types, or rolled their own implementations if they needed it.

**Review by djuber — APPROVED** [review]

File: Gemfile.lock (hunk 1 above)
Comment (jeremyf) [reply Gemfile.lock:870]: When I saw that we were removing Virtus, I was quite happy.  I love Virtus but DryRB is so much nicer (and maintained).

Comment (citizen428) [reply Gemfile.lock:870]: This PR ends up removing 6 gems in total, which I really like. Going through the Rails 7 update process once again confirms that every dependency is a liability. I mean even revisit usage of `store_attribute` for profile fields at one point, as the scope changed we don't have much need for typed accessors as we're only dealing with strings now.

**Review by jeremyf — APPROVED** [review]

**Comment by benhalpern** [comment]
@citizen428 If we can't find use of this, I believe we used in in association with language settings in the past?
